### ISSUE TITLE
Error with connect to rpc

### ISSUE THREAD

**Issue by mraksoll4** [uaktags/explorer#5]
cd /home/ubuntu/explorer && /home/ubuntu/.nvm/src/node-v6.17.1/node scripts/sync.js index update
script launched with pid: 10289
Update Stats Completed.
/home/ubuntu/explorer/lib/database.js:797
              lib.syncLoop(block.tx.length, function (subloop) {
                                   ^

TypeError: Cannot read property 'length' of undefined
    at /home/ubuntu/explorer/lib/database.js:797:36
    at Request._callback (/home/ubuntu/explorer/lib/explorer.js:122:14)
    at Request.self.callback (/home/ubuntu/explorer/node_modules/request/request.js:185:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
    at Request.<anonymous> (/home/ubuntu/explorer/node_modules/request/request.js:1161:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:188:7)
    at IncomingMessage.<anonymous> (/home/ubuntu/explorer/node_modules/request/request.js:1083:12)
    at IncomingMessage.g (events.js:292:16)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:978:12)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)

**Comment by mraksoll4** [comment]
Problem is at RPC connection

**Comment by uaktags** [comment]
Can you get me what "block" is?
Something like "console.log(block); may suffice, at line 796-797
```
if (block && block.name != "RpcError") {
 console.log("Block Output. Line 797:", block);
 lib........
```
Curious as to what the issue is with the RPC.

**Comment by mraksoll4** [comment]
i find how fix this error https://github.com/iquidus/explorer/pull/67/files#diff-2d71e1ae0f109ee656aa6aaad7e342f3
, but  its stil cannot connect to rpc, i think problem at bitcoin-api with bitcoin-core.
Somting wrong with api communication.
 Node version what i was use 6+.

**Comment by mraksoll4** [comment]
all work with node
ubuntu@ubuntu:~/explorer$ node -v
v8.10.0
ubuntu@ubuntu:~/explorer$ npm -v
3.5.2
sorry only 2 issues

Starting cluster with pid: 4632
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed
Unknown Setting: 'language'. This setting doesn't exist or it was removed


and at sync.js
cd /home/ubuntu/explorer && /usr/bin/nodejs scripts/sync.js index update
Unknown Setting: 'language'. This setting doesn't exist or it was removed
script launched with pid: 3294
(node:3294) DeprecationWarning: collection.ensureIndex is deprecated. Use createIndexes instead.
Update Stats Completed.

this fixed by 
useCreateIndex: true
sync.js add forgoted line 129
      mongoose.connect(dbString, { useCreateIndex: true, useNewUrlParser: true }, function(err) {

but now i hv error if stop sync.js and tun again with update

fix this only cleare db table and re-create them.

cd /home/ubuntu/explorer && /usr/bin/nodejs scripts/sync.js index update
Unknown Setting: 'language'. This setting doesn't exist or it was removed
script launched with pid: 4402
Update Stats Completed.
/home/ubuntu/explorer/lib/explorer.js:384
    module.exports.syncLoop(tx.vin.length, function (loop) {
                                   ^

TypeError: Cannot read property 'length' of undefined
    at Object.prepare_vin (/home/ubuntu/explorer/lib/explorer.js:384:36)
    at /home/ubuntu/explorer/lib/database.js:162:15
    at Request._callback (/home/ubuntu/explorer/lib/explorer.js:122:14)
    at Request.self.callback (/home/ubuntu/explorer/node_modules/request/request.js:185:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (/home/ubuntu/explorer/node_modules/request/request.js:1161:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at IncomingMessage.<anonymous> (/home/ubuntu/explorer/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1064:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)

+at explorer bug , when if click to blockhash at tx.
![dccc1](https://user-images.githubusercontent.com/32709596/56692405-cb422180-66ea-11e9-9fac-86d64580a614.jpg)
![dddc](https://user-images.githubusercontent.com/32709596/56692406-cb422180-66ea-11e9-8af1-ad7894a0272c.jpg)

its give 404

**Comment by uaktags** [comment]
Are you missing
```
 // language settings
  "language": "us",
"language_fallback": "us"
```
from your settings? I've added a few parameters during the forks as I worked towards pushing us up higher and incorporate missing functionality. Basically we need that, your lib/settings.js should have:
```
//Multilanguage
exports.languages = ["us"];
exports.language_fallback = "us";
```
and then line 58-59 of app.js should have:
```
    fallbackLng: settings.language_fallback,
preload: settings.language,
```
according to our master branch.

I'll play around with a test script to try to narrow down your issue with the RPC communication.

To confirm, you have the following two issues:

- [x] Unknown Settings 'language'
- [ ] Possible RPC issue during the ``` lib.syncLoop(block.tx.length, function (subloop) { ```

**Comment by uaktags** [comment]
Try to keep the issues singular as best as possible, it gets hard to track progress if we keep adding more and doing inline edits on comments.

**Comment by mraksoll4** [comment]
lang - settings exist
![exist1](https://user-images.githubusercontent.com/32709596/56693034-21fc2b00-66ec-11e9-95ed-dbe86a946974.jpg)
![exist3](https://user-images.githubusercontent.com/32709596/56693037-21fc2b00-66ec-11e9-8bbb-47e0ba422d2b.jpg)
![ddc1](https://user-images.githubusercontent.com/32709596/56693462-312fa880-66ed-11e9-986f-ca815c1c11ff.jpg)

**Comment by uaktags** [comment]
*head...meet desk*
modify settings.js to say exports.language, not languages. I'll make a commit for it shortly. Not sure why that didn't provide an issue for me. Give that a try.

**Comment by mraksoll4** [comment]
after changing 
//Multilanguage
exports.languages = ["us"];
exports.language_fallback = "us";
to
//Multilanguage
exports.language = ["us"];
exports.language_fallback = "us";
```
TypeError: this.options.preload.forEach is not a function
    at I18n.loadResources (/home/ubuntu/explorer/node_modules/i18next/dist/commonjs/i18next.js:234:32)
    at setLng (/home/ubuntu/explorer/node_modules/i18next/dist/commonjs/i18next.js:319:16)
    at I18n.changeLanguage (/home/ubuntu/explorer/node_modules/i18next/dist/commonjs/i18next.js:325:9)
    at Timeout.load [as _onTimeout] (/home/ubuntu/explorer/node_modules/i18next/dist/commonjs/i18next.js:178:16)
    at ontimeout (timers.js:482:11)
    at tryOnTimeout (timers.js:317:5)
    at Timer.listOnTimeout (timers.js:277:5)
/home/ubuntu/explorer/node_modules/i18next/dist/commonjs/i18next.js:234
          this.options.preload.forEach(function (l) {

```

**Comment by uaktags** [comment]
Alright, spinning up a fresh clone. One sec. I'll jump back on, but this is giving us grief over the i18n preload functionality.

**Comment by uaktags** [comment]
hmm, can't seem to reproduce the issue. Have master branch syncing right now on vulc.

**Comment by uaktags** [comment]
Alright @mraksoll4, I may have this fixed now according to #7 . Please test and let me know.

**Comment by uaktags** [comment]
Closing for now.
